assets/javascripts/discourse/widgets/discourse-reactions-actions.js.es6

```diff
@@ -345,54 +355,69 @@ export default createWidget("discourse-reactions-actions", {
           .current_user_used_main_reaction;
         const template = document.createElement("template");
         template.innerHTML = iconHTML(
-          hasUsedMainReaction ? `far-${mainReactionIcon}` : mainReactionIcon
+          hasUsedMainReaction ||
+            (attrs.reaction && attrs.reaction !== mainReactionName)
+            ? `far-${mainReactionIcon}`
+            : mainReactionIcon
         ).trim();
         const icon = template.content.firstChild;
         icon.style.transform = `scale(${scales[1]})`;
 
         mainReaction.parentNode.replaceChild(icon, mainReaction);
         scaleReactionAnimation(icon, scales[1], scales[0], () => {
           this.dropUserReaction();
-          this.addUserReaction(this.siteSettings.discourse_reactions_like_icon);
+          if (attrs.reaction && attrs.reaction !== mainReactionName) {
+            this.addUserReaction(attrs.reaction);
+          } else {
+            this.addUserReaction(
+              this.siteSettings.discourse_reactions_like_icon
+            );
+          }
 
-          if (!this.attrs.post.current_user_reaction) {
+          if (this.attrs.post.current_user_reaction) {
+            this.attrs.post.reaction_users_count -= 1;
+            this.setCurrentUserReaction();
```

Comment:
I find it a little bit weird that we use setCurrentUserReaction with no arg, we don't set anything, we actually unset

Reply:
We can change the method's name. Do you have any suggestions?

Reply:
or we can just modify the name of the current method as `serOrUnsetCurrentUserReaction` ?

Reply:
I think the easiest is to keep setCurrentUserReaction() but pass it null setCurrentUserReaction(null) in this case, it will make it more explicit.

Reply:
Sure, that's too will look nice